src/dmd/typesem.d

```diff
@@ -1156,7 +1156,12 @@ private extern (C++) final class TypeSemanticVisitor : Visitor
         {
             if (s)
             {
-                s.error(loc, "is used as a type");
+                auto td = s.isTemplateDeclaration;
+                if (td && td.onemember && td.onemember.isAggregateDeclaration)
+                    mtype.error(loc, "%s template `%s` is used as a type without instantiation",
```

Comment:
I suggest an improvement:
```
template %s `%s` is used as a type without instantiation; to instantiate it use `%s!(arguments)`
```

Reply:
What if the user has a variable called `arguments` already in scope? They might get confused.

Reply:
@MetaLang Maybe *use `%s!()`* instead? That way if the template requires arguments they will get an error to that effect, and this error may come up when the template doesn't need arguments anyway, as in the forum example.